src/drivers/imu/bmi088/BMI088_accel.cpp

```diff
@@ -0,0 +1,597 @@
+    accel_data[index] = BMI088_ACC_X_L | DIR_READ;
+
+    const hrt_abstime timestamp_sample = hrt_absolute_time();
+
+    if (OK != transfer(accel_data, accel_data, sizeof(accel_data))) {
+        return;
+    }
+
+    check_registers();
+
+    /* Extracting accel data from the read data */
+    index = 2; // Skip the dummy byte at index=1
+    uint16_t lsb, msb, msblsb;
+
+    lsb = (uint16_t)accel_data[index++];
+    msb = (uint16_t)accel_data[index++];
+    msblsb = (msb << 8) | lsb;
+    report.accel_x = (int16_t)msblsb; /* Data in X axis */
+
+    lsb = (uint16_t)accel_data[index++];
+    msb = (uint16_t)accel_data[index++];
+    msblsb = (msb << 8) | lsb;
+    report.accel_y = (int16_t)msblsb; /* Data in Y axis */
+
+    lsb = (uint16_t)accel_data[index++];
+    msb = (uint16_t)accel_data[index++];
+    msblsb = (msb << 8) | lsb;
+    report.accel_z = (int16_t)msblsb; /* Data in Z axis */
+
+    // Extract the temperature data
+    // Note: the temp sensor data is only updated every 1.28s (see "Register 0x22-0x23 Temperature Sensor Data" section in BMI088 Datasheet)
+    index = 0;
+    accel_data[index] = BMI088_ACC_TEMP_H | DIR_READ;
+    // Need to perform a dummy read, hence the num bytes to read is 3 (plus 1 send byte)
+    if (OK != transfer(accel_data, accel_data, 4)) {
+        return;
+    }
+    index = 2;
+    msb = (uint16_t)accel_data[index++];
+    lsb = (uint16_t)accel_data[index++];
+    uint16_t temp = msb*8 + lsb/32;
+    if (temp > 1023) {
+        report.temp = temp - 2048;
+    }
+    else {
+        report.temp = temp;
+    }
+
+
+    if (report.accel_x == 0 &&
+            report.accel_y == 0 &&
+            report.accel_z == 0) {
+        // all zero data - probably a SPI bus error
+        perf_count(_bad_transfers);
+        perf_end(_sample_perf);
+        // note that we don't call reset() here as a reset()
+        // costs 20ms with interrupts disabled. That means if
+        // the bmi088 accel does go bad it would cause a FMU failure,
+        // regardless of whether another sensor is available,
+        return;
+    }
+
+    if (_register_wait != 0) {
+        // we are waiting for some good transfers before using
+        // the sensor again, but don't return any data yet
+        _register_wait--;
+        return;
+    }
+
+    // report the error count as the sum of the number of bad
+    // transfers and bad register reads. This allows the higher
+    // level code to decide if it should use this sensor based on
+    // whether it has had failures
+    const uint64_t error_count = perf_event_count(_bad_transfers) + perf_event_count(_bad_registers);
+    _px4_accel.set_error_count(error_count);
+
+    // Convert the bit-wise representation of temperature to degrees C
+    _accel_last_temperature_copy = (report.temp * 0.125f) + 23.0f;
```

Comment:
Note: This calculation of the temp has an offset of 23 deg C. This comes from section "Register 0x22 – 0x23: Temperature sensor data" of the BMI088 datasheet.